src/components/TableSection.tsx

```diff
@@ -46,31 +38,73 @@ const columns: TableColumnsType<UserType> = [
     key: "phone",
   },
   {
-    title: "Website",
-    dataIndex: "website",
-    key: "website",
-    render: (website) => <span>W: {website}</span>,
+    title: "Date of Birth",
+    dataIndex: "birthDate",
+    key: "birthDate",
+  },
+  {
+    title: "Age",
+    dataIndex: "age",
+    key: "age",
+  },
+  {
+    title: "Gender",
+    dataIndex: "gender",
+    key: "gender",
+  },
+  {
+    title: "Picture",
+    dataIndex: "image",
+    key: "image",
+    render: (image) => <img src={image} />,
   },
 ];
-// const data: UserType[] = UserRecords;
 
 const UserTable: React.FC = () => {
-  const { data = [], isFetching } = useGetUsersQuery();
+  const { data, isFetching } = useGetUsersQuery();
 
-  const normalizedData = useMemo(
-    () =>
-      map(data, (value: UserType, index: number) => ({
+  const options = "YYYY-M-D";
+  const selectedUsers = useSelector((state) => state.users.usernames); //retrieve data from slice
+  const selectedAgeRange = useSelector((state) => state.users.selectedAgeRange);
+  const selectedGender = useSelector((state) => state.users.selectedGenders);
+  const selectedDobDates = useSelector(
+    (state) => state.users.selectedDateOfBirthFilter
+  );
+
+  const filteredUsers = useMemo(() => {
+    return chain(data)
+      .filter((user) =>
+        (size(selectedUsers) > 0
+          ? includes(selectedUsers, user.username)
+          : false) &&
+        (size(selectedGender) > 0
+          ? includes(selectedGender, user.gender)
+          : false) &&
+        (size(selectedAgeRange) > 0
```

Comment:
we can use `lodash ~ inRange` to handle this logic
https://lodash.com/docs/4.17.15#inRange